presto-native-execution/src/test/java/com/facebook/presto/nativeworker/AbstractTestNativeIcebergPositionDeleteQueries.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.nativeworker;
+
+import com.facebook.presto.testing.QueryRunner;
+import com.facebook.presto.tests.AbstractTestQueryFramework;
+import org.testng.annotations.Test;
+
+import static com.facebook.presto.nativeworker.NativeQueryRunnerUtils.createNationWithFormat;
+import static org.testng.Assert.assertEquals;
+
+public abstract class AbstractTestNativeIcebergPositionDeleteQueries
+        extends AbstractTestQueryFramework
+{
+    private final String storageFormat = "PARQUET";
+
+    @Override
+    protected void createTables()
+    {
+        QueryRunner queryRunner = (QueryRunner) getExpectedQueryRunner();
+
+        createNationWithFormat(queryRunner, storageFormat);
+    }
+
+    @Test
+    public void testIcebergReadPositionDeletes()
+    {
+        QueryRunner javaIcebergQueryRunner = (QueryRunner) getExpectedQueryRunner();
+
+        try {
+            javaIcebergQueryRunner.execute("CREATE TABLE iceberg_native_position_delete_test AS SELECT * FROM nation");
+
+            // ASSERT number of rows in the table before delete is executed
+            assertQueryResultCount("SELECT * FROM iceberg_native_position_delete_test", 25);
+
+            // DELETE a ROW and create a position delete file
+            javaIcebergQueryRunner.execute("DELETE FROM iceberg_native_position_delete_test where nationkey = 10");
+
+            // ASSERT number of rows in the table after delete is executed
+            assertQueryResultCount("SELECT * FROM iceberg_native_position_delete_test", 24);
+
+            // DELETE another ROW to create a second delete file
+            javaIcebergQueryRunner.execute("DELETE FROM iceberg_native_position_delete_test where nationkey = 20");
+
+            // ASSERT number of rows in the table after second delete operation
+            // This also tests iceberg read with Multiple Delete Files
+            assertQueryResultCount("SELECT * FROM iceberg_native_position_delete_test", 23);
+        }
+        finally {
+            javaIcebergQueryRunner.execute("DROP TABLE IF EXISTS iceberg_native_position_delete_test");
+        }
+    }
+
```

Comment:
Can you please add test cases on partitioned tables:
1. delete on partition keys
2. delete on non-partition keys

Reply:
Can you please add test cases on tables with NULL values?
1. Delete NULL values
2. Delete non-Null values